plugins/optimization-detective/storage/class-od-url-metrics-post-type.php

```diff
@@ -119,6 +119,10 @@ public static function get_post( string $slug ): ?WP_Post {
 	public static function get_url_metrics_from_post( WP_Post $post ): array {
 		$this_function = __METHOD__;
 		$trigger_error = static function ( string $message, int $error_level = E_USER_NOTICE ) use ( $this_function ): void {
+			// Default to E_USER_NOTICE.
+			if ( ! in_array( $error_level, array( E_USER_NOTICE, E_USER_WARNING, E_USER_ERROR, E_USER_DEPRECATED ), true ) ) {
+				$error_level = E_USER_NOTICE;
+			}
 			wp_trigger_error( $this_function, esc_html( $message ), $error_level );
 		};
```

Comment:
It's too bad that this doesn't seem to work:

```php
		/**
		 * Triggers error.
		 *
		 * @param string $message     The message explaining the error.
		 * @param int    $error_level Optional. The designated error type for this error.
		 *
		 * @phpstan-param E_USER_DEPRECATED|E_USER_ERROR|E_USER_NOTICE|E_USER_NOTICE $error_level
		 */
		$trigger_error = static function ( string $message, int $error_level = E_USER_NOTICE ) use ( $this_function ): void {
			if ( ! in_array( $error_level, array( E_USER_ERROR, E_USER_NOTICE, E_USER_NOTICE, E_USER_DEPRECATED ), true ) ) {
				throw new InvalidArgumentException( 'Invalid error level provided.' );
			}
			wp_trigger_error( $this_function, esc_html( $message ), $error_level );
		};
```

If I do `$trigger_error( 'foo', 181293 )` it doesn't trigger any PHPStan static analysis error.

Reply:
But you do get a "InvalidArgumentException", right?

Reply:
Yeah, I mean more this part:

```
@phpstan-param E_USER_DEPRECATED|E_USER_ERROR|E_USER_NOTICE|E_USER_NOTICE $error_level
```

I'm not getting any static analysis issues when passing an integer other than those.